app/models/ticket.rb

```diff
@@ -56,12 +92,22 @@ def build_jobs
     DEFAULT_LOGGER.info "Building Ticket: #{self.id}"
     self.class.registry.add(self)
     @collated_stream = CollatedStream.new(tag_lines: context.is_a?(Group))
-    self.jobs = nodes.map do |n|
-      Job.new(node: n, ticket: self).tap do |job|
+
+    # Adds Node Base Jobs
+    nodes.map do |n|
+      self.jobs << Job.new(node: n, ticket: self).tap do |job|
```

Comment:
Whilst the previous implementation is "technically fine" it had a code smell to it after `has_context` was implemented.

Previously the `nodes.map` was responsible for assigning an array to the `jobs` attribute even if `has_context` is false. Then an additional "no context job" would be appended to the mapped version of the `nodes` array. This would technically work, but it breaks the standard design pattern for the class.

Instead the `jobs` array is initialised by the attribute definition: 
https://github.com/openflighthpc/flight-action-api/blob/13c2d831e4b3b05386eb1974033d442b4807fd72/app/models/ticket.rb#L75

Because `jobs` is now an array, the "node jobs" and "no context job" should be appended instead of replacing the array.

The useless `map` has now been replaced with `each`:
https://github.com/openflighthpc/flight-action-api/pull/15/commits/9193f06128f9fe7ea362c3b603d482e62cfdebcc